### ISSUE TITLE
Should functions marked as deprecated raise a linting issue?

### ISSUE THREAD

**Issue by seeruk** [golang/lint#375]
I've been refactoring a bunch of code recently, and marking things as deprecated by changing the comments above types, like:

```diff
-// Foo is a foo.
+// Deprecated: Foo is a foo. Use Baz instead.
 type Foo struct {
     bar Bar
 }
```

I also try to make sure I don't have any linting issues on projects where possible; however, in making the above change an issue is raised:

```
./b.go:3:1: comment on exported type Foo should be of the form "Foo ..." (with optional leading article)
```

This introduces a lot of noise and obscures the view of comments that are actually incorrect. Is this the correct way to mark something as deprecated? This sort of ties into another issue I've seen that discusses whether or not linting issues should be raised when a deprecated type is used (as I'd rather see that than the above message).

**Comment by andybons** [comment]
Thanks for the PR.

This appears to be [out of scope](https://github.com/golang/lint/blob/master/README.md#scope) and so is being closed. If you feel this is in error please feel free to re-open.

Please note this is not the forum to discuss edits to [CodeReviewComments](https://golang.org/wiki/CodeReviewComments). If you wish to propose changes to it please send a mail to [golang-nuts](https://groups.google.com/forum/#!forum/golang-nuts).

**Comment by dgryski** [comment]
staticcheck complains about the use of deprecated methods.

**Comment by seeruk** [comment]
Yeah, but so does lint, just not in an easy to interpret manner, sadly. I get a load of noise just because I've deprecated something currently, which makes spotting comments that aren't actually valid more difficult. I'd be happy to see lint simply not complain if a comment was actually valid still, but with `Deprecated: ` in front. i.e. this should be valid, if `T` matches the thing you're adding a comment to:

```go
// Deprecated: T
type T struct {
```

Right now, it'll say that the comment is not in the format `// T...`.

**Comment by dominikh** [comment]
@seeruk The correct way to deprecate something, based on your initial example, is as follows:

```
// Foo is a foo.
//
// Deprecated: Don't use Foo, use Bar.
type Foo struct {
    bar Bar
}
```

That is, the `Deprecated: ...` is its own paragraph, and you don't remove the existing documentation.

**Comment by seeruk** [comment]
Right, that makes a lot of sense! I'll get them fixed up in that case. Thanks dominikh.
